crypto/bio/bio_lib.c

```diff
@@ -45,22 +45,21 @@ int BIO_set(BIO *bio, const BIO_METHOD *method)
     bio->references = 1;
     bio->num_read = 0L;
     bio->num_write = 0L;
-    if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_BIO, bio, &bio->ex_data))
-        return 0;
-
     bio->lock = CRYPTO_THREAD_lock_new();
```

Comment:
one thing to consider is have a static BIO object with initializers and then do an assignment, to get the initializatoin.  If not that, then just doing memset() and setting the non-zero pointers is better.  while you're in this part of the code.